File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count=defaultdict(int)
        res=0
        j=0
        for i in range(len(s)):
            count[s[i]]+=1
            max_freq=max(count.values())
            cur=i-j+1
            if cur-max_freq>k:
                count[s[j]]-=1
                j+=1
            res=max(res,i-j+1)
        return res

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
       HashMap<Character, Integer> count = new HashMap<>();
        int res = 0;
        int maxFreq = 0;
        int j = 0;

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            count.put(c, count.getOrDefault(c, 0) + 1);

            maxFreq = Math.max(maxFreq, count.get(c));
            int windowSize = i - j + 1;

            if (windowSize - maxFreq > k) {
                char leftChar = s.charAt(j);
                count.put(leftChar, count.get(leftChar) - 1);
                j++;
            }

            res = Math.max(res, i - j + 1);
        }

        return res; 
    }
}